packages/flutter_tools/lib/src/base/io.dart

```diff
@@ -92,12 +92,29 @@ ExitFunction _exitFunction = _defaultExitFunction;
 
 /// Exits the process.
 ///
+/// Throws [StateError] if assertions are enabled and the dart:io exit
+/// is still active when called. This may indicate exit was called in
+/// a test without being configured correctly. This behavior can be
+/// removed by setting the `FLUTTER_INTEGRATION_TEST` environment
```

Comment:
Is `FLUTTER_INTEGRATION_TEST` new? Is there any pre-existing one that could work?

Reply:
there is a FLUTTER_TEST value that is set by `flutter test` and for some reason by the test_driver.dart. I could reuse that field since I don't think it is used otherwise for pub run test

Reply:
Switched to FLUTTER_TEST